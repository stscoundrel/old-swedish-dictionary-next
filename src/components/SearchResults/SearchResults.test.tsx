import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import SearchResults from './index'

const words = [
  {
    headword: 'afköra',
    partOfSpeech: 'vb',
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora',
    foundIn: [' , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.'],
  },
  {
    headword: 'afköra 2',
    partOfSpeech: 'vb',
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora',
    foundIn: [' , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.'],
  },
  {
    headword: 'afköra 3',
    partOfSpeech: 'vb',
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora',
    foundIn: [' , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.'],
  },
]

describe('SearchResults component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<SearchResults words={words} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<SearchResults words={words} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
