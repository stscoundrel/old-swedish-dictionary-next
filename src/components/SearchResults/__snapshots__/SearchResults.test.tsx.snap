// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component Matches snapshot 1`] = `
[
  <p
    className="blue"
  >
    3
     results found
  </p>,
  <ul
    className="list"
  >
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/afkora"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          afköra
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              {
                "__html": " , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/afkora"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          afköra 2
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              {
                "__html": " , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/afkora"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          afköra 3
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              {
                "__html": " , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.",
              }
            }
          />
        </ul>
      </div>
    </li>
  </ul>,
]
`;
